Create update_nightscout.sh

Navid200 opened the pull request:
Lets you:
   Update Nightscout
   Install from a fork

Navid200 committed:
Create update_nightscout.sh

Navid200 commented:
@jamorham Would you please change the file permissions to 755?
I need to figure out a reliable way to do this on Windows.

Navid200 committed:
Add files via upload

Navid200 committed:
Add files via upload

Cancel and escape on the second dialog will also terminate the script.
The screen is cleared after each dialog.

Navid200 committed:
Add files via upload

Changed permissions to 775

Navid200 committed:
Add files via upload

Exit if any of the fork parameters are left empty.
Every exit has an echo now.
Removed tar extraction.

Navid200 committed:
Update update_nightscout.sh

Navid200 commented:
![Screenshot 2022-10-10 235124](https://user-images.githubusercontent.com/51497406/194995608-d9fd239e-0763-4713-9a12-32d9ef5d3494.png)

![Screenshot 2022-10-10 235155](https://user-images.githubusercontent.com/51497406/194995739-bc0792b0-a894-41c0-ae97-248ef745520a.png)

jamorham commented:
There seems to be a lot of duplication with the installation script? Can you explain? Can't you just change the git origin of the existing repo position, checkout, pull from there and then execute to rebuild and then reboot to install the changes?

Navid200 commented:
@jamorham 
I needed to install the dialog because I wanted this to be incremental.
After install, I need to post install.  Otherwise, it will not work.

I also have tested every exit and the two installs every time I have made a small change.  Every test takes 20 minutes.  Every change takes about an hour including the two tests.

I started from what I knew would work, which was a full repeat of the install.  And I have been gradually pruning it.
I wanted to get this in as soon as possible so that people could update. 
I can always do clean-up.  

Now that I know what your concern is, I will do major pruning.

Thanks

Navid200 committed:
Add files via upload

The pull now happens in a different directory than /srv/$repo.  It happens in /nightscout_start/$repo.  
Some redundant sections were deleted.

Navid200 commented:
@jamorham 
The git pull is now done in a dedicated directory other than the one the original install was done in.
I have removed some sections that were redusant.
Please let me know if you see anything else that needs improvement.
Thanks

Navid200 committed:
Add files via upload

The script will reboot after the install.

Navid200 committed:
Add files via upload

755 permissions

Navid200 committed:
Add files via upload

New line at the end

Navid200 committed:
Add files via upload

Removed unnecessary commands.
Added more comments.

Navid200 commented:
Everything remaining is necessary.

jamorham commented:
As this is a standalone script I'm not sure whether the instructions for needing to run from installation are correct.

For next steps I would suggest making a menu script which gives the users choices like, change/install freedns, update nightscout, install again, reboot server, copy databases etc for all the utility scripts and then call out to each as requested. Also add a return to shell option and we can then make such a menu spawn when the user logs in and therefore eliminate a lot of the command line actions.

Navid200 commented:
Thanks

I will follow exactly those instructions for the next steps after free I figure out how to set up free dns correctly.
